### PR TITLE
ZEPPELIN-2199: Fix lapply issue in sparkR

### PR DESCRIPTION
**Pull request by vipul1409**
### What is this PR for?
Function createRDDFromArray used for creating R RDD expects a JavaSparkContext object instead of spark context. This PR address that concern. 

### What type of PR is it?
Bug Fix

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-2199

### How should this be tested?
Build Zeppelin and Run 
%r
families <- c("gaussian", "poisson")
df <- createDataFrame(iris)
train <- function(family)
{ 
    model <- glm(Sepal.Length ~ Sepal.Width + Species, iris, family = family) 
    summary(model) 
}
model.summaries <- spark.lapply(families, train)
print(model.summaries)

It fails in current master but will pass in this branch.

### Questions:
* Does the licenses files need update?
No
* Is there breaking changes for older versions?
Not completely sure about this. 
* Does this needs documentation?
No.

### CODE REVIEW

**Comment by vipul1409** [comment]
Can anyone help me understand the build error it looks unrelated to my change,

**Comment by prabhjyotsingh** [comment]
Did you switch on 'zeppelin' repository from  https://travis-ci.org/profile  (https://zeppelin.apache.org/contribution/contributions.html#continuous-integration) ? 

Also after that goto https://travis-ci.org/andreaTP/zeppelin/settings and switch on "Build pushes" and "Build pull requests".

**Comment by prabhjyotsingh** [comment]
I meant https://travis-ci.org/vipul1409/zeppelin/settings instead of "andreaTP". Typo

**Comment by vipul1409** [comment]
I have enabled travis for my fork. At this URL https://travis-ci.org/vipul1409/zeppelin/settings "Build pushes" and "Build pull requests" are enabled. is there any other setting that needs to be enabled.
I have checked contributing.md and followed all the steps.

**Comment by prabhjyotsingh** [comment]
I still don't see any builds on your travis (https://travis-ci.org/vipul1409/zeppelin/branches). 

Can you do a empty commit on  this branch (ZEPPELIN-2199) `git commit --amend;git push -f` ?

**Comment by vipul1409** [comment]
Thanks for your help @prabhjyotsingh . I can see the build started in travis now.

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 1]
```diff
@@ -152,6 +154,15 @@ public SparkContext getSparkContext() {
     }
   }
 
+  public JavaSparkContext getJavaSparkContext() {
+    synchronized (sharedInterpreterLock) {
+      if (jsc == null) {
+        jsc = JavaSparkContext.fromSparkContext(sc);
```
Comment (felixcheung) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: this needs to be cleared along with `sc`, correct?
https://github.com/vipul1409/zeppelin/blob/cebf9701d4be708af27eed3eb786f8bc3d2bf628/spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java#L1435

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java [hunk 2]
```diff
@@ -574,9 +585,9 @@ private void setupConfForSparkR(SparkConf conf) {
         archives = conf.get("spark.yarn.dist.archives");
       }
       if (archives != null) {
-        archives = archives + "," + sparkRPath + "#sparkr";
+        archives = archives + "," + "file:/" + sparkRPath + "#sparkr";
```
Comment (felixcheung) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: is this related to this JIRA?

Comment (vipul1409) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: No this is not.... Will remove it.

**Comment by vipul1409** [comment]
Re ran the failed job and it passed.

File: spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java (hunk 1 above)
Comment (vipul1409) [inline spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java]: Done

**Comment by felixcheung** [comment]
@Leemoonsoo Travis CI everything passes actually - it looks like Jenkins has timed-out while the last suite was still running.

**Comment by felixcheung** [comment]
Do we have unit tests for SparkR? If so it would be good to add a test for this.
Otherwise looks good to me

**Comment by vipul1409** [comment]
There is not SparkRInterpreterTest.java, looks like there are no test for SparkR.

**Comment by vipul1409** [comment]
Can this PR be merged?

**Comment by felixcheung** [comment]
it looks like it passed with this build https://travis-ci.org/vipul1409/zeppelin/builds/207896434

**Comment by felixcheung** [comment]
merging if no more comment

**Comment by felixcheung** [comment]
@vipul1409 do you have a user ID in Apache JIRA? We could resolve this issue to you.

**Comment by felixcheung** [comment]
merged to master and branch-0.7

**Comment by vipul1409** [comment]
@felixcheung My user id is vipul.1409@gmail.com
